eras/conway/impl/src/Cardano/Ledger/Conway/Governance/Proposals.hs

```diff
@@ -238,33 +239,51 @@ proposalsAddAction ::
   GovActionState era ->
   Proposals era ->
   Maybe (Proposals era)
-proposalsAddAction gas ps = withGovActionParent gas (Just psWithGas) update
+proposalsAddAction gas ps = checkInvariantAfterAddition gas ps <$> runProposalsAddAction gas ps
+
+-- | Add a single @`GovActionState`@ to the @`Proposals`@ forest.
+-- The tree to which it is added is picked according to its
+-- @`GovActionPurpose`@.
+unsafeProposalsAddAction ::
+  forall era.
+  HasCallStack =>
+  GovActionState era ->
+  Proposals era ->
+  Proposals era
+unsafeProposalsAddAction gas ps =
```

Comment:
It would be nice to have such functions in tests instead, but I guess to do that we'd have to expose `runProposalsAddAction`, which would be confusing since it has the same signature as  `proposalsAddAction` . 
Should we add at least some comment saying that it's for testing only , or the `unsafe` prefix is expressive enough ?

Reply:
Yes, we can move this function to `unsafeMkProposals` instead. I would hesitate to expose enough to re-implement this functionality elsewhere, that makes changing things in the future more annoying. But I'm not firm either way.